### ISSUE TITLE
feat(solvers): expose spectral solvers as lineax operators and gaussx preconditioners (`as_lineax_operator`, `as_preconditioner`) + variable-coefficient CG docs example

### ISSUE THREAD

**Issue by jejjohnson** [jejjohnson/spectraldiffx#112]
Part of #84. Works with the currently pinned gaussx v0.0.17; no gaussx change required. Cheapest, highest-value item in the epic.

## Summary

A constant-coefficient spectral Helmholtz solve is an excellent preconditioner for variable-coefficient elliptic problems (`∇·(a∇ψ) − λψ = f`, spatially varying Coriolis/stratification in QG, immersed boundaries, …), and gaussx already has the machinery to use one: `gaussx.linear_solve(op, b, solver=CGSolver(...), preconditioner=OperatorPreconditioner(M_inv))`. Nothing in spectraldiffx exposes its solvers in that form, so users have to write the `lineax.AbstractLinearOperator` wrapper themselves.

## Measured (128², a ∈ [0.5, 1.5], λ = 1, float64, CPU, jitted)

| Method | iterations | time |
|---|---|---|
| plain `lineax.CG` on the FD stencil, rtol 1e-6 | 549 | 144 ms |
| CG with the spectral solve as preconditioner via `lineax` `options={"preconditioner": M_inv}` | 16 | 14.1 ms |
| same via `gaussx.linear_solve(..., preconditioner=OperatorPreconditioner(M_inv))` | 16 | 12.3 ms |
| the spectral solve itself wrapped as a custom `lx.AbstractLinearSolver` | – | 0.485 ms (0.97x of the direct function) |
| `jax.grad` through the lineax wrapper (implicit differentiation) | – | 1.14 ms vs 0.64 ms direct (1.8x) |

Note `gaussx.PreconditionedCGSolver` is the partial-Cholesky preconditioner, not bring-your-own; the bring-your-own path is `OperatorPreconditioner`.

## Proposed design

On the Fourier Helmholtz solver classes (today `SpectralHelmholtzSolver*`, `MixedBCHelmholtzSolver*`; after #107 `HelmholtzSolver`), and on the Chebyshev/spherical solvers once #79 and #81 land:

```python
def as_lineax_operator(self, alpha=None) -> lx.AbstractLinearOperator:
    """The forward operator (∇² − α) as a lineax operator (FunctionLinearOperator or a small custom class with static grid)."""

def as_inverse_operator(self, alpha=None) -> lx.AbstractLinearOperator:
    """The solve as a lineax operator: mv(f) == self.solve(f, alpha)."""

def as_preconditioner(self, alpha=None) -> gaussx.OperatorPreconditioner:
    """Convenience wrapper around as_inverse_operator for gaussx.linear_solve / CGSolver."""
```

Plus a custom `lx.AbstractLinearSolver` (`SpectralSolver`) so `lineax.linear_solve(A, b, solver=SpectralSolver())` dispatches to the direct transform solve when `A` is one of these operators (this is the 0.97x path). Requires the static-fields fix (#102) so the operators are hashable pytrees.

Docs: a new page `docs/preconditioning_guide.md` with the variable-coefficient example above (executed via the doc-snippet test from #95), and a note that direct functions remain faster for hot loops and for gradients.

## Acceptance criteria

- [ ] `as_inverse_operator().mv(f)` equals `solve(f)` to 1e-13; `as_lineax_operator().mv(as_inverse_operator().mv(f))` equals `f` to 1e-10 on the retained modes.
- [ ] Variable-coefficient test: preconditioned CG converges in ≤ 25 iterations at 128² where plain CG needs > 400.
- [ ] Works under `jax.jit` and `jax.grad` w.r.t. `f` (implicit diff through `lineax`).
- [ ] The guide page builds and its snippets execute in CI.
